File: demo/php/trade_demo.php
<?php

$url = "https://open-api.bingx.com";
$apiKey = "Set your api key here!!";
$secretKey = "Set your secret key here!!";

function getOriginString(array $params) {
    // combine origin string
    $originString = "";
    $first = true;
    foreach($params as $n => $v) {
          if (!$first) {
              $originString .= "&";
          }
          $first = false;
          $originString .= $n . "=" . $v;
    }
    return $originString;
}

function getSignature(string $originString) {
    global $secretKey;
    $signature = hash_hmac('sha256', $originString, $secretKey, true);
    $signature = bin2hex($signature);
    return $signature;
}

function getRequestUrl(string $path, array $params) {
    global $url;
    $requestUrl = $url.$path."?";
    $first = true;
    foreach($params as $n => $v) {
          if (!$first) {
              $requestUrl .= "&";
          }
          $first = false;
          $requestUrl .= $n . "=" . $v;
    }
    return $requestUrl;
}

function httpPost($url)
{
    global $apiKey;
    $curl = curl_init($url);
    curl_setopt($curl, CURLOPT_POST, true);
    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($curl, CURLOPT_USERAGENT, "curl/7.80.0");
    curl_setopt($curl, CURLOPT_HTTPHEADER, array(
        "X-BX-APIKEY:".$apiKey,
    ));
    $response = curl_exec($curl);
    curl_close($curl);
    return $response;
}

function placeMarketOrder(string $symbol, string $side, string $quantity, string $positionSide) {

    // interface info
    $path = "/openApi/swap/v2/trade/order";

    // interface params
    $params = array();
    $params['symbol'] = $symbol;
    $params['side'] = $side;
    $params['type'] = "MARKET";
    $params['quantity'] = $quantity;
    $params['positionSide'] = $positionSide;
    $date = new DateTime();
    $params['timestamp'] = $date->getTimestamp()*1000;

    // generate signature
    $originString = getOriginString($params);
    $signature = getSignature($originString);
    $params["signature"] = $signature;

    // send http request
    $requestUrl = getRequestUrl($path, $params);
    $result = httpPost($requestUrl);
    echo "\t";
    echo $result;
    echo "\n";
}

echo "placeMarketOrder:\n";
placeMarketOrder("LINK-USDT", "BUY", "5", "LONG");

?>

File: demo/java/java-demo/TradeDemo.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLConnection;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.sql.Timestamp;
import java.util.Map;
import java.util.TreeMap;

public class TradeDemo {
    String url = "https://open-api.bingx.com";
    String apiKey = "Set your api key here!!";
    String secretKey = "Set your secret key here!!";

    private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray();

    public static String bytesToHex(byte[] bytes) {
        char[] hexChars = new char[bytes.length * 2];
        for (int j = 0; j < bytes.length; j++) {
            int v = bytes[j] & 0xFF;
            hexChars[j * 2] = HEX_ARRAY[v >>> 4];
            hexChars[j * 2 + 1] = HEX_ARRAY[v & 0x0F];
        }
        return new String(hexChars);
    }

    byte[] hmac(String algorithm, byte[] key, byte[] message) throws NoSuchAlgorithmException, InvalidKeyException {
        Mac mac = Mac.getInstance(algorithm);
        mac.init(new SecretKeySpec(key, algorithm));
        return mac.doFinal(message);
    }


    String generateHmac256(String message) {
        try {
            byte[] bytes = hmac("HmacSHA256", secretKey.getBytes(), message.getBytes());
            return bytesToHex(bytes);
        } catch (Exception e) {
            System.out.println("generateHmac256 expection:" + e);
        }
        return "";
    }


    String getMessageToDigest(String method, String path, TreeMap<String, String> parameters) {
        Boolean first = true;
        String valueToDigest = "";
        for (Map.Entry<String, String> e : parameters.entrySet()) {
            if (!first) {
                valueToDigest += "&";
            }
            first = false;
            valueToDigest += e.getKey() + "=" + e.getValue();
        }
        return valueToDigest;
    }

    String getRequestUrl(String path, String parameters) {
        String urlStr = url + path + "?" + parameters;
        return urlStr;
    }

    void send_request(String requestUrl, String method) {
        try {
            URL url = new URL(requestUrl);
            URLConnection conn = url.openConnection();
            HttpURLConnection http = (HttpURLConnection)conn;
            http.setRequestMethod(method); // PUT is another valid option
            http.addRequestProperty("X-BX-APIKEY", apiKey);
            http.setDoOutput(true);
            conn.setDoOutput(true);
            conn.setDoInput(true);

            String result = "";
            String line = "";
            BufferedReader in = new BufferedReader(
                    new InputStreamReader(conn.getInputStream()));
            while ((line = in.readLine()) != null) {
                result += line;
            }

            System.out.println("\t"+result);

        } catch (Exception e) {
            System.out.println("expection:" + e);
        }
    }


    void placeMarketOrder(String symbol, String side, String positionSide, String quantity) {
        String method = "POST";
        String path = "/openApi/swap/v2/trade/order";
        String timestamp = ""+new Timestamp(System.currentTimeMillis()).getTime();

        TreeMap<String, String> parameters = new TreeMap<String, String>();
        parameters.put("symbol", symbol);
        parameters.put("side", side);
        parameters.put("type", "MARKET");
        parameters.put("positionSide", positionSide);
        parameters.put("quantity", quantity);

        parameters.put("timestamp", timestamp);

        String valueToDigest = getMessageToDigest(method, path, parameters);
        String messageDigest = generateHmac256(valueToDigest);
        String parametersString = valueToDigest + "&signature=" + messageDigest;
        String requestUrl = getRequestUrl(path, parametersString);

        send_request(requestUrl, method);
    }

    public static void main(String[] args) {
        TradeDemo h = new TradeDemo();

        System.out.println("placeMarketOrder:");
        h.placeMarketOrder("LINK-USDT", "BUY", "LONG", "5");
    }
}
